product.go

```diff
@@ -34,22 +34,23 @@ type ProductServiceOp struct {
 	client *Client
 }
 
-type productStatus string
+// ProductStatus represents a Shopify product status.
+type ProductStatus string
```

Comment:
Do you need to make this public?

Reply:
Sadly yes because when you set the `ProductListOptions` you need to import this type if you want to set a slice.

e.g: 
```go
options := goshopify.ProductListOptions{
   Status: []ProductStatus{"active", "draft"}, // ProductStatus type
}

products, pages, err := client.Product.ListWithPagination(options)
if err != nil {
  return nil, "", err
}
```

Reply:
I see, makes sense, thanks!